code/tracing/MainFrameTimer.cpp

```diff
@@ -0,0 +1,37 @@
+//
+//
+
+#include "MainFrameTimer.h"
+
+#include "tracing/scopes.h"
+#include "tracing/categories.h"
+
+namespace tracing {
+
+MainFrameTimer::MainFrameTimer() : _out("profiling.csv") {
+}
+MainFrameTimer::~MainFrameTimer() {
+	_out.close();
+}
+void MainFrameTimer::processEvent(const trace_event* event) {
+	if (event->scope != &MainFrameScope || event->category != &MainFrame) {
+		return;
+	}
+
+	switch(event->type) {
+		case EventType::AsyncBegin:
+			_begin_time = event->timestamp;
+		case EventType::AsyncEnd:
```

Comment:
Intentional fall-through? This would output a message at the beginning of the event with a message like "2400; 0"